mcode/mcode-coverage-checker
# Adding assessment coverage checking + testing and fixes for treatment coverage checking

# Description
Issue: STEAM-847

This PR adds coverage checking for Assessment resources, adds testing for Treatment coverage checking, fixes issues found with Treatment coverage checking and implements the new ID enum files for all other sections 

## Important Changes
Coverage checking has been added for the assessment section of the mCODE diagram in `assessmentCoverage.js`

Tests added for Assessment coverage checking in `assessmentCoverage.test.js` and accompanying `assessmentBundle.json` test bundle created.

Additionally, I decided to add tests for the Treatment coverage checking in `treatmentCoverage.test.js` (creating an accompanying `treatmentBundle.json`) since they didn't exist. While doing this I noticed a few issues with treatment coverage checking and decided to fix them here as well. Most notably, many of the coverage checking functions skipped the "covered" property all the other coverage objects used.

I also added each other section to the `coverageProfileIds.js` and `coverageSectionIds.js` files that were introduced with the previous Treatment Coverage PR

## Testing Recommendations
- For Both Assessment and Treatment:
  - Ensure that new tests cover sufficient cases and pass
  - Check that appropriate fields are checked for
  - Make sure that each field has a proper true/false value when it is covered or not
- Make sure the other sections still work as expected with new use of profile and section IDs

# Checklists

**Submitter:**

@ACCT1 :
